modules/docs/sc_common.md

```diff
@@ -593,3 +600,119 @@ local wrong_pattern = "a random pattern %2"
 local result = test_common:is_valid_pattern(wrong_pattern)
 --> result is: false
 ```
+
+## sleep method
+
+The **sleep** method make the script pause for the given amount of seconds.
+
+### sleep: parameters
+
+| parameter                              | type   | optional | default value |
+| -------------------------------------- | ------ | -------- | ------------- |
+| the number of seconds you need to wait | number | no       |               |
+
+### sleep: example
+
+```lua
+local seconds_to_wait = 10
+
+test_common:sleep(seconds_to_wait)
+```
+
+## create_sleep_counter_table method
+
+The **create_sleep_counter_table** method creates a table to handle sleep counters. Useful when you want to log something less often after a few times.
+
+This table will contain a min, a max, a value and a step number that will be used to increment the value. The table is also shipped with three methods
+
+| method name      | purpose                                                                                                                          | params | returns                                                   |
+| ---------------- | -------------------------------------------------------------------------------------------------------------------------------- | ------ | --------------------------------------------------------- |
+| reset()          | set the counter value to the minimum value                                                                                       |        |                                                           |
+| is_max_reached() | checks if the counter is equal or above the max                                                                                  |        | true or false if the counter is equal or above to the max |
+| increment()      | increase the counter by adding to it the step number                                                                             |        |                                                           |
+| sleep()          | will use the sc_common:sleep() function using the counter value as an argument and it will automatically increment() the counter |        |                                                           |
```

Comment:
```suggestion
| sleep()          | will use the sc_common:sleep() function using the counter value as an argument, it will automatically increment() the counter |        |                                                           |
```